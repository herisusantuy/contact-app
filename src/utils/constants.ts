export const testID = {
  swipeableWrapper: "card-contact-swipeable-wrapper",
  photoProfile: "card-contact-photo-profile",
  fullnameText: "card-contact-fullname-text",
  ageText: "card-contact-age-text",
  pressableView: "floating-button-pressable-view",
  plusIcon: "floating-button-plus-icon",
  textInput: "form-text-field",
  customModal: "custom-modal",
  cancelButton: "cancel-button",
  saveButton: "save-button",
};
