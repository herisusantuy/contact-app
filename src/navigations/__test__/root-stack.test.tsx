import * as React from "react";
import { screen, render, fireEvent } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";

import RootStackNavigator from "../root-stack";
import { renderWithProviders } from "../../utils/test-utils";

test("Show Home screen.", () => {
  const tree = renderWithProviders(<RootStackNavigator />).toJSON();
  expect(tree).toMatchSnapshot();
});
