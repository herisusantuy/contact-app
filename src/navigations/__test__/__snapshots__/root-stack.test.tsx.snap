// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Home screen. 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              {
                "backgroundColor": "white",
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <Modal
              animationType="none"
              deviceHeight={null}
              deviceWidth={null}
              hardwareAccelerated={false}
              hideModalContentWhileAnimating={false}
              onBackdropPress={[Function]}
              onModalHide={[Function]}
              onModalWillHide={[Function]}
              onModalWillShow={[Function]}
              onRequestClose={[Function]}
              panResponderThreshold={4}
              scrollHorizontal={false}
              scrollOffset={0}
              scrollOffsetMax={0}
              scrollTo={null}
              statusBarTranslucent={false}
              supportedOrientations={
                [
                  "portrait",
                  "landscape",
                ]
              }
              swipeThreshold={100}
              transparent={true}
              visible={false}
            />
            <ActivityIndicator
              style={
                {
                  "marginTop": 20,
                }
              }
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "green",
                  "borderColor": "green",
                  "borderRadius": 100,
                  "borderWidth": 1,
                  "bottom": 100,
                  "height": 70,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 30,
                  "width": 70,
                }
              }
              testID="floating-button-pressable-view"
            >
              <Text />
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Contacts"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
